railties/test/commands/server_test.rb

```diff
@@ -132,12 +136,28 @@ def test_restart_command_contains_customized_options
     args = ["-p", "4567"]
     ARGV.replace args
 
-    options = Rails::Server::Options.new.parse! args
-    server = Rails::Server.new options
+    options = parse_arguments(args)
     expected = "bin/rails server -p 4567"
 
-    assert_equal expected, server.default_options[:restart_cmd]
+    assert_equal expected, options[:restart_cmd]
   ensure
     ARGV.replace original_args
   end
+
+  private
+
+    def parse_arguments(args)
+      Rails::Command::ServerCommand.class_eval do
```

Comment:
Not sure, I had to do something similar for the other commands. But if we can make it happen, great 👍